x-pack/plugins/elastic_assistant/server/routes/attack_discovery/helpers.ts

```diff
@@ -434,9 +434,9 @@ export const findAttackDiscoveryByConnectorId = async ({
     return null;
   }
 
+  // update lastViewedAt time as this is the function used for polling by connectorId
   return dataClient.updateAttackDiscovery({
     attackDiscoveryUpdateProps: {
-      status: attackDiscovery.status,
```

Comment:
@andrew-goldstein this was the cause of the race condition that perpetuated the running state. 

In `x-pack/plugins/elastic_assistant/server/ai_assistant_data_clients/attack_discovery/update_attack_discovery.ts` we run the es update API and then refetch the document. This polling function must have ran right between a success update and a success refetch, resetting the state to `running`.

Reply:
so `findAttackDiscoveryByConnectorId` on L428 returns `status: running`, the success update happens and `status: succeeded`, this update runs and switches it back to `status: running`, then the success update refetches and receives `status: running`

Reply:
🐎